tools/clang/lib/CodeGen/CGHLSLMS.cpp

```diff
@@ -720,11 +720,9 @@ DxilResourceProperties CGMSHLSLRuntime::BuildResourceProperty(QualType resTy) {
   } break;
   case DXIL::ResourceClass::Sampler: {
     DxilSampler::SamplerKind kind = KeywordToSamplerKind(RD->getName());
```

Comment:
Is RD canonical type decl?  If not, it could be typedef and wrong for this usage.

Reply:
const RecordType *RT = resTy->getAs<RecordType>(); will remove the typedef.